url026791d51b_dhleong_SeriesTracker_v6_tgz-pJ8-stv6_SeriesTrackerJ8/src/stv6/Profile.java

```diff
@@ -0,0 +1,801 @@
+		if (!r.getGetVars().isSet("id")) 
+			return null;
+		
+		// read the ID defensively and load the series
+		try {
+			int seriesId = Integer.parseInt( r.getGetVars().getValue("id") );			
+			return (seriesId < 0) ? null : db.getSeriesAsUser(seriesId, series, r.getUser());
+		} catch (NumberFormatException e) {
+			return null;
+		}
+		
+	}
+	
+	/**
+	 * Note that this is NOT cached!
+	 * @return A SeriesList of all Series (as TrackedSeries)
+	 * 	available from the selected EpisodeManager 
+	 */
+	public SeriesList getAvailableSeries() {
+		return epmgr.getAvailableSeries();
+	}
+	
+	/**
+	 * @return A File pointing to the executable
+	 * 	with which to execute plugins
+	 */
+	public File getPluginExe() {
+		return pluginExe;
+	}
+	
+	/**
+	 * @return The port number ST should listen on
+	 */
+	public int getPort() {
+		return port;
+	}
+
+
+	public File[] getPossibleProfiles() {
+		return possibleProfiles;
+	}
+
+	public List<User> getPossibleUsers() {
+		return possibleUsers;
+	}
+	
+	/**
+	 * @return A SeriesList of all Series attached
+	 * 	to this profile
+	 */
+	public SeriesList getSeries() {
+		return series;
+	}
+
+	/**
+	 * @see Database#getSyncNewSeries(SyncSettings)
+	 */
+	public SeriesList getSyncNewSeries() { 
+		return db.getSyncNewSeries(syncSettings);
+	}
+
+	public List<TrackData> getSyncNewTracks() {
+		return db.getSyncNewTracks(syncSettings);
+	}
+
+	public List<User> getSyncNewUsers() {
+		return db.getSyncNewUsers(syncSettings);
+	}
+
+	public boolean fillUser(Request r) {
+		if (globalUser != null) {
+			r.setUser( globalUser );
+			return true;
+		}
+		
+		r.readHeaders();
+		if (r.getCookies().isSet(STClient.COOKIE_USERID)) {
+			// load the user. this is super lame, but I'm lazy :P
+			String strId;
+			for (User u : possibleUsers) {
+				strId = String.valueOf(u.getId());
+				if (r.getCookies().getValue(STClient.COOKIE_USERID).equals(strId)) {
+					r.setUser(u);
+					return true;
+				}
+			}
+		}
+		
+		return false;
+	}
+	
+	public static Profile getInstance() {
+		return instance_;
+	}
+
+	public String getTemplatePathFor(Request r, String fileName) {
+		// get path from command line
+		
+		// check for GET var "template"
+		if (r.getGetVars().isSet("template"))
+			return buildPath(TEMPLATE_PATH, r.getGetVars().getValue("template"), fileName);
+		
+		// get template name from command line
+		if (templateName != null)
+			return buildPath(TEMPLATE_PATH, templateName, fileName);
+			
+		// just default
+		return buildPath(TEMPLATE_PATH, fileName);
+	}
+
+	/**
+	 * @return True if there are multiple profiles to pick from
+	 */
+	public boolean hasMultipleProfiles() {
+		return possibleProfiles != null && possibleProfiles.length > 1;
+	}
+
+	/**
+	 * @return True if there are multiple users to pick from
+	 */
+	public boolean hasMultipleUsers() {
+		return possibleUsers != null && possibleUsers.size() > 1;
+	}
+	
+	/**
+	 * @param args The ones from main()
+	 */
+	public void initialize(String[] args) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 281-288, lines 290-300, lines 310-329 into separate methods.